jerry-ext/context/include/jerryscript-ext/context.h

```diff
@@ -0,0 +1,55 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef JERRYX_CONTEXT_H
+#define JERRYX_CONTEXT_H
+
+#include "jerryscript.h"
+
+/**
+ * Function to pass to jerry_init_with_user_context() as its second parameter
+ */
+void *jerryx_context_init (void);
+
+/**
+ * Function to pass to jerry_init_with_user_context() as its third parameter
+ */
+void jerryx_context_deinit (void *user_context_p);
+
+typedef struct
+{
+  /** the initialization function */
+  jerry_user_context_init_cb init_cb;
+  /** the deinitialization function */
+  jerry_user_context_deinit_cb deinit_cb;
+  /** the index of the slot */
+  int index;
```

Comment:
Do you still need the runtime assignment of the index? Do you need the index at all?

The index can be written as a constant that is known at link-time:
```(&__jerryx_context_slot_myslot - &__start_jerryx_context_slots)```

This way the `jerryx_context_slot_t` can also be `static const` instead of just `static`.